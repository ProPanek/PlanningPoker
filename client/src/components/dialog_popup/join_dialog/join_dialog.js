import React, { Component } from 'react'
import Dialog from '@material-ui/core/Dialog'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import TextField from '@material-ui/core/TextField'
import DialogActions from '@material-ui/core/DialogActions'
import Button from '@material-ui/core/Button'
import { inject, observer } from 'mobx-react'
import { withRouter } from 'react-router-dom'
import { decorate, observable } from 'mobx'
import routes from 'routes'

class JoinDialogComponent extends Component {
  constructor (props) {
    super(props)
    this.userName = ''
    this.roomPassword = ''
  }

  componentDidMount () {}

  handleChangUserName = e => {
    this.userName = e.target.value
  }

  handleChangRoomPassword = e => {
    this.roomPassword = e.target.value
  }
  joinRoom = () => {
    const {
      match: { params },
      store: {
        roomStore: { joinRoom, setRoomPassword, setRoomId },
        socketStore: { openNotification },
        userStore: { setUserName, userName, loggedIn }
      }
    } = this.props
    if (loggedIn ? userName : this.userName && this.roomPassword) {
      if (!loggedIn) {
        setUserName(this.userName)
        setRoomId(params.id)
      }
      setRoomPassword(this.roomPassword)
      joinRoom(
        loggedIn ? userName : this.userName,
        params.id,
        this.roomPassword
      )
    } else {
      openNotification('To join a room you need to fill all inputs', 'warning')
    }
  }

  cancelJoinRoom = () => {
    const {
      store: {
        userStore: { loggedIn }
      },
      history: { push }
    } = this.props
    loggedIn ? push(routes.rooms()) : push(routes.root())
  }

  render () {
    const {
      store: {
        userStore: { isConnected, loggedIn }
      }
    } = this.props
    return (
      <Dialog open={!isConnected} aria-labelledby='form-dialog-title'>
        <DialogTitle id='form-dialog-title'>Join Room</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Please write your username below if you want to join to the room.
          </DialogContentText>
          {!loggedIn && (
            <TextField
              autoFocus
              margin='dense'
              id='userName'
              label='User Name'
              type='text'
              fullWidth
              value={this.userName}
              onChange={this.handleChangUserName}
            />
          )}
          <TextField
            autoFocus
            margin='dense'
            id='roomPassword'
            label='Room Password'
            type='password'
            fullWidth
            value={this.roomPassword}
            onChange={this.handleChangRoomPassword}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={this.cancelJoinRoom} color='primary'>
            Cancel Join
          </Button>
          <Button onClick={this.joinRoom} color='primary' variant='contained'>
            Join Room
          </Button>
        </DialogActions>
      </Dialog>
    )
  }
}

decorate(JoinDialogComponent, {
  userName: observable,
  roomPassword: observable
})

export { JoinDialogComponent }
export default inject('store')(withRouter(observer(JoinDialogComponent)))
