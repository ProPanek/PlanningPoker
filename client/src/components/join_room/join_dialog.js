import React, {Component} from 'react'
import Dialog from '@material-ui/core/Dialog'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import TextField from '@material-ui/core/TextField'
import DialogActions from '@material-ui/core/DialogActions'
import Button from '@material-ui/core/Button'
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import {decorate, observable} from "mobx";


class JoinDialog extends Component {
  handleChange = (e) => {
    this.input = e.target.value
  }
  joinRoom = (name) => {
    const {match: {params}} = this.props
    this.props.store.joinRoom(params.id, params.password, name)
    this.props.store.openJoinDialog = false
    this.props.store.connected = true
  }

  cancelJoinRoom = () => {
    this.props.history.push(`/`)
  }

  render() {

    return (
      <Dialog
        open={this.props.store.openJoinDialog}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Join Room</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Please write your username below if you want to join to the room.
          </DialogContentText>
          <TextField
            autoFocus
            margin="dense"
            id="name"
            label="User Name"
            type="text"
            fullWidth
            value={this.input}
            onChange={this.handleChange}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={this.cancelJoinRoom} color="primary">
            Cancel Join
          </Button>
          <Button onClick={() => {
            this.joinRoom(this.input)
          }} color="primary" variant="contained">
            Join Room
          </Button>
        </DialogActions>
      </Dialog>
    )
  }
}

decorate(JoinDialog, {
  input: observable
});
export default inject("store")(withRouter(observer(JoinDialog)));

// export default JoinDialog