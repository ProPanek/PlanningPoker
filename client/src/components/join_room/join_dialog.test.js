import React from "react";
import {JoinDialog} from "./join_dialog";
import sinon from 'sinon'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


describe("CreateRoom", () => {

  it("renders without crashing", () => {
    const store = {
     openJoinDialog: true
    }
    const component = shallow(<JoinDialog store={store}/>)
    expect(component).toMatchSnapshot()
  });

  it("cancels joining room", () => {
    const props = {
      store: {openJoinDialog: true},
      history: { push: sinon.spy() },
      match: { params: {} }
    }

    const component = shallow(<JoinDialog {...props}/>)
    const instance = component.instance()
    instance.cancelJoinRoom()
    expect(props.history.push.called).toBe(true)

  });

  it("joins room", () => {
    const props = {
      store: {openJoinDialog: true, joinRoom: sinon.spy(), user:{connected: true}},
      match: { params: {} }
    }

    const component = shallow(<JoinDialog {...props}/>)
    const instance = component.instance()
    instance.joinRoom()
    expect(props.store.joinRoom.called).toBe(true)

  });
});