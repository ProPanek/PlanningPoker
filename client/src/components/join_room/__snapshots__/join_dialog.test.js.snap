// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRoom renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JoinDialog
    store={
      Object {
        "openJoinDialog": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-labelledby": "form-dialog-title",
      "children": Array [
        <WithStyles(DialogTitle)
          id="form-dialog-title"
        >
          Join Room
        </WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)>
            Please write your username below if you want to join to the room.
          </WithStyles(DialogContentText)>
          <TextField
            autoFocus={true}
            fullWidth={true}
            id="userName"
            label="User Name"
            margin="dense"
            onChange={[Function]}
            required={false}
            select={false}
            type="text"
            variant="standard"
          />
          <TextField
            autoFocus={true}
            fullWidth={true}
            id="roomPassword"
            label="Room Password"
            margin="dense"
            onChange={[Function]}
            required={false}
            select={false}
            type="text"
            variant="standard"
          />
        </WithStyles(DialogContent)>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
          >
            Cancel Join
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            Join Room
          </WithStyles(Button)>
        </WithStyles(DialogActions)>,
      ],
      "open": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Join Room",
          "id": "form-dialog-title",
        },
        "ref": null,
        "rendered": "Join Room",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(DialogContentText)>
              Please write your username below if you want to join to the room.
            </WithStyles(DialogContentText)>,
            <TextField
              autoFocus={true}
              fullWidth={true}
              id="userName"
              label="User Name"
              margin="dense"
              onChange={[Function]}
              required={false}
              select={false}
              type="text"
              variant="standard"
            />,
            <TextField
              autoFocus={true}
              fullWidth={true}
              id="roomPassword"
              label="Room Password"
              margin="dense"
              onChange={[Function]}
              required={false}
              select={false}
              type="text"
              variant="standard"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Please write your username below if you want to join to the room.",
            },
            "ref": null,
            "rendered": "Please write your username below if you want to join to the room.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoFocus": true,
              "fullWidth": true,
              "id": "userName",
              "label": "User Name",
              "margin": "dense",
              "onChange": [Function],
              "required": false,
              "select": false,
              "type": "text",
              "value": undefined,
              "variant": "standard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoFocus": true,
              "fullWidth": true,
              "id": "roomPassword",
              "label": "Room Password",
              "margin": "dense",
              "onChange": [Function],
              "required": false,
              "select": false,
              "type": "text",
              "value": undefined,
              "variant": "standard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              Cancel Join
            </WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
              variant="contained"
            >
              Join Room
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cancel Join",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Cancel Join",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Join Room",
              "color": "primary",
              "onClick": [Function],
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Join Room",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-labelledby": "form-dialog-title",
        "children": Array [
          <WithStyles(DialogTitle)
            id="form-dialog-title"
          >
            Join Room
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)>
              Please write your username below if you want to join to the room.
            </WithStyles(DialogContentText)>
            <TextField
              autoFocus={true}
              fullWidth={true}
              id="userName"
              label="User Name"
              margin="dense"
              onChange={[Function]}
              required={false}
              select={false}
              type="text"
              variant="standard"
            />
            <TextField
              autoFocus={true}
              fullWidth={true}
              id="roomPassword"
              label="Room Password"
              margin="dense"
              onChange={[Function]}
              required={false}
              select={false}
              type="text"
              variant="standard"
            />
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              Cancel Join
            </WithStyles(Button)>
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
              variant="contained"
            >
              Join Room
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Join Room",
            "id": "form-dialog-title",
          },
          "ref": null,
          "rendered": "Join Room",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(DialogContentText)>
                Please write your username below if you want to join to the room.
              </WithStyles(DialogContentText)>,
              <TextField
                autoFocus={true}
                fullWidth={true}
                id="userName"
                label="User Name"
                margin="dense"
                onChange={[Function]}
                required={false}
                select={false}
                type="text"
                variant="standard"
              />,
              <TextField
                autoFocus={true}
                fullWidth={true}
                id="roomPassword"
                label="Room Password"
                margin="dense"
                onChange={[Function]}
                required={false}
                select={false}
                type="text"
                variant="standard"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Please write your username below if you want to join to the room.",
              },
              "ref": null,
              "rendered": "Please write your username below if you want to join to the room.",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "fullWidth": true,
                "id": "userName",
                "label": "User Name",
                "margin": "dense",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "text",
                "value": undefined,
                "variant": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "fullWidth": true,
                "id": "roomPassword",
                "label": "Room Password",
                "margin": "dense",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "text",
                "value": undefined,
                "variant": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
              >
                Cancel Join
              </WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                Join Room
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cancel Join",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cancel Join",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Join Room",
                "color": "primary",
                "onClick": [Function],
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Join Room",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
